### PR TITLE
Add documentation for OARO flowsheet

### PR DESCRIPTION
**Pull request by luohezhiming**
## Fixes/Resolves:

(replace this with the issue #1219 fixed or resolved, if no issue exists then a brief statement of what this PR does)

## Summary/Motivation:
OARO flowsheet documentaion

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/technical_reference/flowsheets/oaro.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
+Osmotically Assisted Reverse Osmosis
+====================================
+
+Introduction
+------------
+
+Osmotically assisted reverse osmosis (OARO) is a nonevaporative membrane-based desalination technology that can treat
```
Comment (zacharybinger) [inline docs/technical_reference/flowsheets/oaro.rst]:
```suggestion
Osmotically assisted reverse osmosis (OARO) is a non-evaporative membrane-based desalination technology that can treat
```

File: docs/technical_reference/flowsheets/oaro.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+high-salinity brines. Compared to conventional reverse osmosis (RO), a saline sweep is added to reduce the pressure
```
Comment (zacharybinger) [inline docs/technical_reference/flowsheets/oaro.rst]:
```suggestion
high-salinity brines. Compared to conventional reverse osmosis (RO), a saline sweep is added to reduce the osmotic pressure
```

File: docs/technical_reference/flowsheets/oaro.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+and the diluted sweep solution will flow into the feed-side of the next stage of OARO module.
+Similarly, the resulting concentrate flows out of the feed-side, gets energy recovered by the ERD and
+pressurized by the recycle pump, and flows back into permeate-side of the last stage.
+Meantime, the diluted sweep solution in the permeate-side flows into the next stage until finally into a conventional RO module.
+Costing relationships for each of the unit models is described in the
+`WaterTAP Costing Package <https://watertap.readthedocs.io/en/latest/technical_reference/costing/watertap_costing.html>`_
+The flowsheet relies on the following key assumptions:
+
+   * supports steady-state only
+   * supports optimization minimizes levelized cost of water (LCOW) with constraints
```
Comment (zacharybinger) [inline docs/technical_reference/flowsheets/oaro.rst]:
```suggestion
   * supports optimization and minimizes levelized cost of water (LCOW) with constraints
```

File: docs/technical_reference/flowsheets/oaro.rst (hunk 1 above)
Comment (zacharybinger) [reply docs/technical_reference/flowsheets/oaro.rst]: Could also describe it as a "pressure-driven" membrane-based desalination technology

**Review by zacharybinger — APPROVED** [review]

File: docs/technical_reference/flowsheets/oaro.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
+Osmotically Assisted Reverse Osmosis
+====================================
+
+Introduction
+------------
+
+Osmotically assisted reverse osmosis (OARO) is a non-evaporative membrane-based desalination technology that can treat
+high-salinity brines. Compared to conventional reverse osmosis (RO), a saline sweep is added to reduce the osmotic pressure
+difference across the membrane as well as enhance water transport. This OARO flowsheet include numbers of
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/oaro.rst]:
```suggestion
difference across the membrane as well as enhance water transport. This OARO flowsheet includes numbers of
```

File: docs/technical_reference/flowsheets/oaro.rst [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+After pressurized by the primary pump, the incoming feed water enters the feed-side of the first OARO module,
+then the water permeates across the membrane and the resulting concentrated brine flows out of the system from the feed-side,
+In addition, the permeate water will dilute a saline sweep solution in the permeate-side
+and the diluted sweep solution will flow into the feed-side of the next stage of OARO module.
+Similarly, the resulting concentrate flows out of the feed-side, gets energy recovered by the ERD and
+pressurized by the recycle pump, and flows back into permeate-side of the last stage.
+Meantime, the diluted sweep solution in the permeate-side flows into the next stage until finally into a conventional RO module.
+Costing relationships for each of the unit models is described in the
+`WaterTAP Costing Package <https://watertap.readthedocs.io/en/latest/technical_reference/costing/watertap_costing.html>`_
+The flowsheet relies on the following key assumptions:
```
Comment (MarcusHolly) [inline docs/technical_reference/flowsheets/oaro.rst]:
```suggestion
Figure 1 shows the process flow diagram for an n-stage OARO system.
After pressurized by the primary pump, the incoming feed water enters the feed-side of the first OARO module,
then the water permeates across the membrane and the resulting concentrated brine flows out of the system from the feed-side,
In addition, the permeate water will dilute a saline sweep solution in the permeate-side
and the diluted sweep solution will flow into the feed-side of the next stage of OARO module.
Similarly, the resulting concentrate flows out of the feed-side, gets energy recovered by the ERD and
pressurized by the recycle pump, and flows back into permeate-side of the last stage.
Meanwhile, the diluted sweep solution in the permeate-side flows into the next stage and finally into a conventional RO module.
Costing relationships for each of the unit models is described in the
`WaterTAP Costing Package <https://watertap.readthedocs.io/en/latest/technical_reference/costing/watertap_costing.html>`_ .
The flowsheet relies on the following key assumptions:
```

**Review by MarcusHolly — COMMENTED** [review]
A few minor comments. I could approve this, but would rather have one of the other reviewers with more background/insight on OARO give the final approval.

**Review by adam-a-a — APPROVED** [review]

**Pull request merged**